reconcile/typed_queries.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any
+
+from reconcile.gql_queries.ocp import ocp_auth
+from reconcile.gql_queries.saas_files import saas_files_small_with_provider
+from reconcile.gql_queries.saas_files import saas_files_full
+from reconcile.utils import gql
+
+
+def get_query_data(query_name: str) -> dict[Any, Any]:
+    gqlapi = gql.get_api()
```

Comment:
Slightly of topic: can we try to get rid of using get_api if we refactor this? Having this singleton makes it hard to handle i.e. shared sessions to speed up queries.
I would love to see these as methods of a Gql Client class. Where we create instances of the Gql Client per thread.